### PR TITLE
Align credentials to type 9- part 1

### PR DESCRIPTION
**Pull request by maimorag**
## Status
- [x] In Progress

## Related Issues
related: https://jira-hq.paloaltonetworks.local/browse/CIAC-5274?filter=-1

## Description
update the integration to type9:
- Thinkst Canary
- workday
- Cisco-umbrella-enforcement

## Minimum version of Cortex XSOAR
- [x] 6.5.0

## Does it break backward compatibility?
- [x]  No

### CODE REVIEW

File: Packs/Cisco-umbrella-enforcement/Integrations/CiscoUmbrellaEnforcement/CiscoUmbrellaEnforcement.py [hunk 1]
```diff
@@ -180,7 +180,7 @@ def test_module(client: Client) -> str:
 def main():
     params = demisto.params()
     base_url = f"{params.get('url')}/1.0/"
-    api_key = params.get('api_key')
+    api_key = params.get('cred_api_key', {}).get('password') or params.get('api_key', None)
```
Comment (Shellyber) [inline Packs/Cisco-umbrella-enforcement/Integrations/CiscoUmbrellaEnforcement/CiscoUmbrellaEnforcement.py:183]: What about a check to see if this key was given? no need?

File: Packs/Cisco-umbrella-enforcement/Integrations/CiscoUmbrellaEnforcement/CiscoUmbrellaEnforcement.yml [hunk 2]
```diff
@@ -12,6 +12,13 @@ configuration:
   name: api_key
   required: true
```
Comment (Shellyber) [inline Packs/Cisco-umbrella-enforcement/Integrations/CiscoUmbrellaEnforcement/CiscoUmbrellaEnforcement.yml:13]: Should this still be required?

**Review by Shellyber — COMMENTED** [review]
See my comments regarding all integrations please

File: Packs/Cisco-umbrella-enforcement/Integrations/CiscoUmbrellaEnforcement/CiscoUmbrellaEnforcement.py [hunk 3]
```diff
@@ -155,7 +155,7 @@ def domain_delete_command(client: Client, args: dict) -> CommandResults:
         if isinstance(curr_context, list):
             curr_context = curr_context[0]
         curr_context['IsDeleted'] = True
-    if response and int(response.status_code) == 204:  # type: ignore
+    if response and int(response.get('status_code')) == 204:  # type: ignore
```
Comment (Shellyber) [inline Packs/Cisco-umbrella-enforcement/Integrations/CiscoUmbrellaEnforcement/CiscoUmbrellaEnforcement.py:158]: Why this change?

File: Packs/Workday/Integrations/Workday/Workday.py [hunk 4]
```diff
@@ -297,10 +297,12 @@ def main():
     params = demisto.params()
     user: str = params.get('credentials', {}).get('identifier')
     base_url: str = params.get('base_url', "").rstrip('/')
-    tenant_name: str = params.get('tenant_name')
+    tenant_name: str = params.get('cred_tenant_name', {}).get('identifier') or params.get('tenant_name', None)
     username = f"{user}@{tenant_name}"
     password: str = params.get('credentials', {}).get('password')
-    token = params.get('token')
+    token = params.get('cred_tenant_name', {}).get('password') or params.get('token')
+    if ((not token) or (not tenant_name)) or ((not user) or (not password)):
```
Comment (Shellyber) [inline Packs/Workday/Integrations/Workday/Workday.py:304]:
```suggestion
    if not (token and tenant_name) or not (user and not password):
```

File: Packs/Workday/Integrations/Workday/Workday.yml [hunk 5]
```diff
@@ -10,16 +10,24 @@ configuration:
   type: 0
 - display: Username
   name: credentials
-  required: true
+  required: false
```
Comment (Shellyber) [inline Packs/Workday/Integrations/Workday/Workday.yml:13]: Should this be changed?

File: Packs/Cisco-umbrella-enforcement/Integrations/CiscoUmbrellaEnforcement/CiscoUmbrellaEnforcement.py (hunk 3 above)
Comment (maimorag) [reply Packs/Cisco-umbrella-enforcement/Integrations/CiscoUmbrellaEnforcement/CiscoUmbrellaEnforcement.py]: undo changes + fixed test

File: Packs/Workday/Integrations/Workday/Workday.py [hunk 6]
```diff
@@ -297,10 +297,12 @@ def main():
     params = demisto.params()
     user: str = params.get('credentials', {}).get('identifier')
     base_url: str = params.get('base_url', "").rstrip('/')
-    tenant_name: str = params.get('tenant_name')
+    tenant_name: str = params.get('cred_tenant_name', {}).get('identifier') or params.get('tenant_name', None)
     username = f"{user}@{tenant_name}"
     password: str = params.get('credentials', {}).get('password')
-    token = params.get('token')
+    token = params.get('cred_tenant_name', {}).get('password') or params.get('token')
+    if not (token and tenant_name) or not (user and not password):
```
Comment (Shellyber) [inline Packs/Workday/Integrations/Workday/Workday.py:304]:
```suggestion
    if not (token and tenant_name) or not (user and password):
```

File: Packs/Workday/Integrations/Workday/Workday.yml [hunk 7]
```diff
@@ -14,12 +14,20 @@ configuration:
...
   type: 4
+  hidden: true
+- display: Tenant name
+  name: cred_tenant_name
+  required: false
+  type: 9
+  displaypassword: token
+  hiddenusername: true
```
Comment (Shellyber) [inline Packs/Workday/Integrations/Workday/Workday.yml:30]: Why hiddenusername: true? the identifier here is tenant_name

Comment (maimorag) [reply Packs/Workday/Integrations/Workday/Workday.yml:34]: fixed

**Review by Shellyber — APPROVED** [review]
